atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/pue/ResilientCommitTimestampPutUnlessExistsTable.java

```diff
@@ -79,8 +100,32 @@ public ListenableFuture<Map<Long, Long>> get(Iterable<Long> cells) {
                 MoreExecutors.directExecutor());
     }
 
+    private synchronized FollowUpAction touchAndReturn(CellInfo cellAndValue) {
```

Comment:
What all is this `synchronized` concurrency controlling? Do we expect contention under load for cache misses?

Reply:
This is so we don't have parallel CAS requests for different start timestamps. Since they could still be in the same Cassandra row, we'd contend with ourselves

Reply:
So this means the CASes will queue up and execute serially. I don't think this is the worst outcome (definitely better than on develop, and you have benchmarked this), but still a bit leery of this

Reply:
We could put it behind a lock based on the row they are actually trying to CAS to. Will also benchmark that